### PR TITLE
Fix missing operation lock for BLE notifications

### PR DESCRIPTION
**Pull request by bdraco**

### CODE REVIEW

File: aiohomekit/controller/ble/pairing.py [hunk 1]
```diff
@@ -638,6 +640,7 @@ def _async_get_service_signature_char(self) -> Characteristic | None:
 
     async def _async_set_broadcast_encryption_key(self) -> None:
         """Get the broadcast key for the accessory."""
+        assert self._operation_lock.locked(), "Should be called with operation lock"
```
Comment (bdraco) [inline aiohomekit/controller/ble/pairing.py:643]:
```suggestion
        assert self._operation_lock.locked(), "_operation_lock should be locked"
```

**Review by Jc2k — APPROVED** [review]

File: aiohomekit/controller/ble/pairing.py [hunk 2]
```diff
@@ -1055,6 +1059,9 @@ async def _async_restore_subscriptions(self) -> None:
             self._restore_pending = False
             return
 
+        assert (
+            self._operation_lock.locked()
+        ), "Should be called under the operation lock"
```
Comment (bdraco) [inline aiohomekit/controller/ble/pairing.py:1064]:
```suggestion
        ), "_operation_lock should be locked"
```

**Comment by bdraco** [comment]
Noticed this since the subscribe to gatt notify causing a disconnect in the middle of a competing op that shouldn't have been competing 😨

**Comment by bdraco** [comment]
10 restarts.  No more random failures due to gatt notify in the middle of a sync 👍

**Comment by bdraco** [comment]
Pushing a tag.  Last one before I have to go to the airport and fly to europe

**Pull request merged**